### ISSUE TITLE
Accessible name: Consider empty attributes as empty name

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#3511]
This was brought up in ACT: https://github.com/act-rules/act-rules.github.io/issues/1861

```html
<input type="image" alt="" title="hello world">
```

The above example in Firefox gets `hello world` as its accessible name, but in Chrome the name is empty, and so axe-core should fail this example.

Since the question's going to come up, this is **different** from how aria-label works. If you put an empty aria-label, Chrome will go onto the next name calculation method. It does not do that for the empty alt attribute case. We'll need to look at other scenarios here too. I don't know how this works for `img` elements for example, or `input[type=button]` with an empty `value`.

Before we can work on this we'll need to research similar cases, figure out where else this kind of logic applies.
